server/server.go

```diff
@@ -351,6 +344,26 @@ func (s *Server) Kill(connectionID uint64, query bool) {
 	}
 }
 
+// GracefulDown waits all clients to close.
+func (s *Server) GracefulDown() {
+	log.Info("graceful shut down.")
+	s.rwlock.RLock()
+	for _, conn := range s.clients {
+		conn.killed = true
+		atomic.StoreInt64(&conn.keepAlive, 42)
```

Comment:
What does 42 means?

Reply:
How about read the original value and compare it later?

Reply:
[42: The answer to life, the universe and everything](https://www.quora.com/Why-and-how-is-42-the-answer-to-life-the-universe-and-everything)

> How about read the original value and compare it later?

Need a extra variable to store the original value.